### PR TITLE
Fix problem with parsing links containing mailto: prefix followed by an email

### PR DESCRIPTION
**Pull request by Skalakid**
This PR fixes potential problems with parsing links containing mailto: prefix followed by an email. The changes were added to enhance parsing inside Live Markdown Preview.
The problem was mainly with parsing `[mailto:someone@example.com](mailto:someone@example.com)`. ExpensiMark instead of one link returns two separate <a> tags inside brackets: 
```
[<a href=\"mailto:someone@example.com\">someone@example.com</a>](mailto:<a href=\"mailto:someone@example.com\">someone@example.com</a>)
```

To fix the problem we decided to collapse such links, similarly like it is done after writing 
`[someone@example.com](someone@example.com)`. 

So now following strings:
- `[someone@example.com](mailto:someone@example.com)`
- `[mailto:someone@example.com](someone@example.com)`
- `[mailto:someone@example.com](mailto:someone@example.com)`

will be collapsed into: `<a href=\"mailto:someone@example.com\">someone@example.com</a>`

https://github.com/Expensify/expensify-common/assets/39538890/c7ed7e97-94d9-4153-b1ae-7bd821c1e3de

### Fixed Issues
$ https://github.com/Expensify/App/issues/31181

# Tests
To test these changes I've added proper tests connected to all needed behaviors for Live Markdown Preview. To test it by yourself you should:
1. Run the NewDot/App
2. Open any chat
3. Send message that contains:
```
[someone@example.com](someone@example.com)
[someone@example.com](mailto:someone@example.com)
[mailto:someone@example.com](someone@example.com)
[mailto:someone@example.com](mailto:someone@example.com)
```
4. Verify if every string is collapsed into just someone@example.com

# QA
Same as tests

### CODE REVIEW

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -60,15 +60,20 @@ export default class ExpensiMark {
...
                     if (g1.match(CONST.REG_EXP.EMOJIS) || !g1.trim()) {
                         return match;
                     }
-                    return `<a href="mailto:${g2}">${g1.trim()}</a>`;
+                    let label = g1.trim();
+                    const href = `mailto:${g2}`;
+                    if (label === href) {
+                        label = g2;
+                    }
+                    return `<a href="${href}">${label}</a>`;
```
Comment (tomekzaw) [inline lib/ExpensiMark.js:76]: I don't like the fact that we mutate `label` here but honestly I cannot think of anything better that wouldn't require duplicating `<a href="...">` string.

**Review by tomekzaw — COMMENTED** [review]
Looks good to me. Thanks for adding proper unit tests.

**Comment by Skalakid** [comment]
@jasperhuangg bump

**Review by jasperhuangg — APPROVED** [review]
Sorry for the delay on this was OOO

**Pull request merged**